metricbeat/helper/conversion.go

```diff
@@ -7,75 +7,96 @@ is returned.
 package helper
 
 import (
+	"fmt"
 	"strconv"
 
+	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 )
 
+type Conv struct {
+	Func Convertor
+	Key  string
+}
+
+type Convertor func(key string, data map[string]string) (interface{}, error)
+
+func AddConversions(event common.MapStr, data map[string]string, conversions common.MapStr) {
+	for key, conversion := range conversions {
+		switch conversion.(type) {
+		case Conv:
+			conv := conversion.(Conv)
+			value, err := conv.Func(conv.Key, data)
+			if err != nil {
+				logp.Err("Error on field '%s': %v", key, err)
+			} else {
+				event[key] = value
+			}
+		case common.MapStr:
+			subEvent := common.MapStr{}
+			AddConversions(subEvent, data, conversion.(common.MapStr))
+			event[key] = subEvent
+		}
+	}
+}
```

Comment:
Thanks for the schema API ideas, I adopted most of them. Please have another look.